### PR TITLE
DOCS-854: Document implementation of a modular arm 

### PR DESCRIPTION
**Pull request by sguequierre**
* Add "custom arm" page to modular-resources/examples

### CODE REVIEW

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+title: "Create a custom Arm model"
+linkTitle: "Custom Arm"
+weight: 100
+type: "docs"
+description: "Implement your own robot arm driver. Code a module with the Go or Python SDKs that provides a custom model of arm as a modular resource."
+tags: ["motor", "odrive", "canbus", "serial", "module", "modular resources", "Python", "python SDK", "CAN"]
+# SMEs: Nicole Jung
+---
+
+The RDK provides a number of built-in {{< glossary_tooltip term_id="model" text="models" >}} that implement the API protocol of the [arm](/components/arm) subtype of component, such as the `ur5e`, `xArm6`, and `xArm7`.
```
Comment (sguequierre) [inline docs/extend/modular-resources/examples/custom-arm.md:11]:
```suggestion
The RDK provides a number of built-in {{< glossary_tooltip term_id="model" text="models" >}} that implement the API protocol of the [arm](/components/arm/) subtype of component, such as the `ur5e`, `xArm6`, and `xArm7`.
```

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 2]
```diff
@@ -0,0 +1,258 @@
+---
+title: "Create a custom Arm model"
```
Comment (JessamyT) [inline docs/extend/modular-resources/examples/custom-arm.md:2]:
```suggestion
title: "Create a Custom Arm Model"
```

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+Create a new directory.
+Give it the name your want you call your custom {{< glossary_tooltip term_id="model" text="model" >}} of arm.
+Save this file as <file>your-model-name.json</file> inside of this directory.
+While completing the following step, make sure to save any new files that you generate inside of this directory.
+
+## Create a custom arm model as a modular resource
+
+To create a custom arm model, code a module in Python with the module support libraries provided by [Viam's SDKs](/program/apis/):
+
+1. [Code a new resource model](#code-a-new-resource-model) implementing all methods the Viam RDK requires in the API definition of its built-in subtype (ex. `rdk:component:arm`).
```
Comment (JessamyT) [inline docs/extend/modular-resources/examples/custom-arm.md:74]:
```suggestion
1. [Code a new resource model](#code-a-new-resource-model) implementing all methods the Viam RDK requires in the API definition of its built-in subtype (for example, `rdk:component:arm`).
```

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+Follow [this guide](/extend/modular-resources/create/#code-a-new-resource-model) and select **Go** to learn how to code a module providing a new arm model in Go.
+
+{{% alert title="Modules vs. modular resources" color="tip" %}}
+
+The Viam module system allows you to integrate custom {{< glossary_tooltip term_id="resource" text="resources" >}} ([components](/components/) and [services](/services/)) into any robot running on Viam.
+
+A configured *module* can make one or more *modular resources* available for configuration.
+
+{{% /alert %}}
+
```
Comment (JessamyT) [inline docs/extend/modular-resources/examples/custom-arm.md:46]:
```suggestion
```
I think this clutters the page and is sufficiently covered in the other pages. I think if people don't understand this, they should go to the modular resources concepts page etc.

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+If you have a robot arm whose native software is not provided with a built-in driver, you probably don't want to replace that arm **($$$)**, but you might want to use the [Arm API](/components/arm/#api) to [program](/program/) and control your arm, or want to use it with a Viam [service](/services/).
+
+Follow these instructions to implement a model of [arm component](/components/arm/) that is not built-in to the RDK:
+
+- [Get your arm's kinematics file](#get-your-arms-kinematics-file)
+- [Create a custom arm model as a modular resource](#create-a-custom-arm-model-as-a-modular-resource)
+  - [Code a new resource model](#code-a-new-resource-model)
+  - [Code a main entry point program](#code-a-main-entry-point-program)
+  - [Compile the module into an executable](#compile-the-module-into-an-executable)
+- [Configure the module and modular resource on your robot](#configure-the-module-and-modular-resource-on-your-robot)
```
Comment (JessamyT) [inline docs/extend/modular-resources/examples/custom-arm.md:34]:
```suggestion
```
I think this is redundant with the right nav and with the rest of the page. I think having all these links can make folks think they need to click them all and expect to jump around to other pages. I think users will be less overwhelmed and will find things faster if this whole intro section becomes shorter, and they can just get straight into the material.

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+
+To create a custom arm model, code a module in Python with the module support libraries provided by [Viam's SDKs](/program/apis/):
+
+1. [Code a new resource model](#code-a-new-resource-model) implementing all methods the Viam RDK requires in the API definition of its built-in subtype (ex. `rdk:component:arm`).
+Import your custom model and API into the main program and register the new resource model with your chosen SDK.
+
+2. [Code a main program](#code-a-main-entry-point-program) that starts the module after adding your desired resources from the registry.
+This main program is the "entry point" to your module.
+
+3. [Compile or package](#compile-the-module-into-an-executable) the module into a single executable that can receive a socket argument from Viam, open the socket, and start the module at the entry point.
```
Comment (JessamyT) [inline docs/extend/modular-resources/examples/custom-arm.md:80]: Similarly to above, I think putting the info for a given step in one relevant place is better here. I'd recommend putting the contents of "1. ...." into the "Code a new resource model" section, perhaps with some edits to blend it with the contents there, and same for each of the other steps. That way, people understand what they're doing when they're actually doing it, and don't have to pop up and down the page for context.

I can see the value of what you’re doing here providing some sort of high level outline. Maybe just shorten all these up a bit and move most of the content down as relevant.

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+``` sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+sudo chmod +x <FILEPATH>/<FILENAME>
+```
+
+Ensure any dependencies for your module (including the [Python SDK](https://python.viam.dev/)) are installed on your robot's computer.
+Your executable will be run by `viam-server` as root, so dependencies need to be available to the root user.
+
+## Configure the module and modular resource on your robot
+
+Follow [these configuration instructions](/extend/modular-resources/configure/) to add your custom resource to your robot.
```
Comment (JessamyT) [inline docs/extend/modular-resources/examples/custom-arm.md:258]:
```suggestion
Follow [these configuration instructions](/extend/modular-resources/configure/) to add your custom resource to your robot.

```
Does this need an end line for linting?

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+description: "Implement your own robot arm driver. Code a module with the Go or Python SDKs that provides a custom model of arm as a modular resource."
+tags: ["motor", "odrive", "canbus", "serial", "module", "modular resources", "Python", "python SDK", "CAN"]
+# SMEs: Nicole Jung
+---
+
+The RDK provides a number of built-in {{< glossary_tooltip term_id="model" text="models" >}} that implement the API protocol of the [arm](/components/arm/) subtype of component, such as the `ur5e`, `xArm6`, and `xArm7`.
+
+{{% alert title="Info" color="info" %}}
+
+*Built-in* indicates each of these models has a driver for their native software in the RDK.
```
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/extend/modular-resources/examples/custom-arm.md:15]: Good w/ me!

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+
+- [Get your arm's kinematics file](#get-your-arms-kinematics-file)
+- [Create a custom arm model as a modular resource](#create-a-custom-arm-model-as-a-modular-resource)
+  - [Code a new resource model](#code-a-new-resource-model)
+  - [Code a main entry point program](#code-a-main-entry-point-program)
+  - [Compile the module into an executable](#compile-the-module-into-an-executable)
+- [Configure the module and modular resource on your robot](#configure-the-module-and-modular-resource-on-your-robot)
+
+This guide uses Viam's Python SDK to code the arm module, but if you want to use the Go Client SDK, you can.
+Follow [this guide](/extend/modular-resources/create/#code-a-new-resource-model) and select **Go** to learn how to code a module providing a new arm model in Go.
```
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/extend/modular-resources/examples/custom-arm.md:37]: Leaving up until I implement the alert but I prefer this oriiginal phrasing technically! i think just saying modular doesn't tell them what they're doing enough

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 10]
```diff
@@ -0,0 +1,258 @@
...
+
+- The RDK's `viam-server` will work with <file>URDF</file> (United Robot Description Format) kinematics files, which are currently the standard for ROS drivers.
+Many <file>URDF</file> "robot descriptions" for industrial robot arm models can be found on [GitHub](https://github.com/) that could work with the arm on Viam.
+
+**Create your own kinematics file:**
+
+- Follow the instructions on [Configure Complex Kinematic Chains](/internals/kinematic-chain-config/) to write a file detailing the geometry of your arm.
+  - Use the [Spatial Vector Algebra (SVA)](/internals/kinematic-chain-config/#kinematic-parameters) kinematic parameter type.
+  - Define the parameters in a </file>.json</file> file.
+  - Following the Frame System's guide to [Configure a Reference Frame](/services/frame-system/frame-config/) when working out the relative [orientation](/internals/orientation-vector/) of the `"links"` on your arm.
```
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/extend/modular-resources/examples/custom-arm.md:62]: hm idk if it needs to but it helped me visually to find that in the frame system JSON?

**Review by andf-viam — APPROVED** [review]
This is great! Buncha small language suggestions, all optional. Thanks!

File: docs/extend/modular-resources/examples/custom-arm.md (hunk 5 above)
Comment (JessamyT) [reply docs/extend/modular-resources/examples/custom-arm.md:34]: I don't understand--you have a linter that autopopulates a TOC?

File: docs/extend/modular-resources/examples/custom-arm.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (JessamyT) [reply docs/extend/modular-resources/examples/custom-arm.md:37]: I see what you mean Sierra regarding more info, but I think the "providing a" wording makes the sentence more complex/awk/hard to parse. What if you put "new modular arm" but with a link to modular resources, or add the glossary tooltip for module?

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 11]
```diff
@@ -0,0 +1,252 @@
...
+
+To create a custom arm model, code a module in Python with the module support libraries provided by [Viam's SDKs](/program/apis/):
+
+1. [Code a new resource model](#code-a-new-resource-model) implementing all methods the Viam RDK requires in the API definition of its built-in subtype (for example, `rdk:component:arm`).
+Import your custom model and API into the main program and register the new resource model with your chosen SDK.
+
+2. [Code a main program](#code-a-main-entry-point-program) that starts the module after adding your desired resources from the registry.
+This main program is the "entry point" to your module.
+
+3. [Compile or package](#compile-the-module-into-an-executable) the module into a single executable that can receive a socket argument from Viam, open the socket, and start the module at the entry point.
```
[the remark this answers is not in the document]
Comment (JessamyT) [reply docs/extend/modular-resources/examples/custom-arm.md:73]: If `viam-server` is accurate, generally best not to talk about RDK if we can say `viam-server` instead

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 12]
```diff
@@ -0,0 +1,252 @@
...
+
+- `viam-server` will work with <file>URDF</file> (United Robot Description Format) kinematics files, which are currently the standard for ROS drivers.
+Many <file>URDF</file> "robot descriptions" for industrial robot arm models can be found on [GitHub](https://github.com/) that could work with the arm on Viam.
+
+**Create your own kinematics file:**
+
+- Follow the instructions on [Configure Complex Kinematic Chains](/internals/kinematic-chain-config/) to write a file detailing the geometry of your arm.
+  - Use the [Spatial Vector Algebra (SVA)](/internals/kinematic-chain-config/#kinematic-parameters) kinematic parameter type.
+  - Define the parameters in a </file>.json</file> file.
+  - Follow the Frame System's guide to [Configure a Reference Frame](/services/frame-system/frame-config/) when working out the relative [orientation](/internals/orientation-vector/) of the `"links"` on your arm.
```
Comment (JessamyT) [inline docs/extend/modular-resources/examples/custom-arm.md:55]:
```suggestion
  - Follow the Frame System's guide to [Configure a Reference Frame](/services/frame-system/frame-config/) when working out the relative [orientations](/internals/orientation-vector/) of the `links` on your arm.
```
[opt] "links" could be completely unstyled because they are just...things? concepts? Not sure how to put it. I mean they're not just code entities. But per your comment on my original suggestion here, I think if you want them styled in some way to call them out, they don't need both quotes and backticks.
[nit] Orientations is plural here I believe

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 13]
```diff
@@ -0,0 +1,258 @@
...
+The Viam module system allows you to integrate custom {{< glossary_tooltip term_id="resource" text="resources" >}} ([components](/components/) and [services](/services/)) into any robot running on Viam.
+
+A configured *module* can make one or more *modular resources* available for configuration.
+
+{{% /alert %}}
+
+## Get your arm's kinematics file
+
+The way arms move across space is more complicated than Viam's other [components](/components/).
+Because of this, the arm, unlike other components, needs a [kinematic configuration file](/internals/kinematic-chain-config/) describing the geometry of the robot arm to work with the RDK's built-in [Motion Service](/services/motion/), on top of the client-inaccessible and mostly static [Frame System Service](/services/frame-system/).
```
[the remark this answers is not in the document]
Comment (JessamyT) [reply docs/extend/modular-resources/examples/custom-arm.md]: I stand by my unedited suggestion :D

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 14]
```diff
@@ -0,0 +1,252 @@
...
+---
+
+The RDK provides a number of built-in {{< glossary_tooltip term_id="model" text="models" >}} that implement the API protocol of the [arm](/components/arm/) subtype of component, such as the `ur5e`, `xArm6`, and `xArm7`.
+
+{{% alert title="Info" color="info" %}}
+
+*Built-in* models each have a software driver in the RDK.
+For example, the `ur5e`'s driver is defined in the RDK as found on [GitHub](https://github.com/viamrobotics/rdk/blob/main/components/arm/universalrobots/ur.go).
+
+Each of these models also has to have a [kinematics file](/internals/kinematic-chain-config/) that specifies the relative [orientation](/internals/orientation-vector/) of links and joints in its kinematic chain, which is located in the RDK in the same directory as the respective driver.
```
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/extend/modular-resources/examples/custom-arm.md:18]: great suggestion thank you!!

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 15]
```diff
@@ -0,0 +1,252 @@
...
+## Get your arm's kinematics file
+
+The way arms move across space is more complicated than Viam's other [components](/components/).
+Because of this, an arm, unlike other components, requires a [kinematic configuration file](/internals/kinematic-chain-config/) describing its geometry.
+This provides necessary information for the built-in [Motion Service](/services/motion/) to work with the arm, on top of the client-inaccessible and mostly static [Frame System Service](/services/frame-system/).
+
+**Find a pre-built kinematics file:**
+
+- `viam-server` will work with <file>URDF</file> (United Robot Description Format) kinematics files, which are currently the standard for ROS drivers.
+Many <file>URDF</file> "robot descriptions" for industrial robot arm models can be found on [GitHub](https://github.com/) that could work with the arm on Viam.
```
Comment (sguequierre) [inline docs/extend/modular-resources/examples/custom-arm.md:48]:
```suggestion
You can find <file>URDF</file> "robot descriptions" for many industrial robot arm models on GitHub that are compatible with the Viam arm component.
```

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 16]
```diff
@@ -0,0 +1,250 @@
...
+
+## Get your arm's kinematics file
+
+The way arms move through space is more complicated than Viam's other [components](/components/).
+Because of this, an arm, unlike other components, requires a [kinematic configuration file](/internals/kinematic-chain-config/) describing its geometry.
+This provides necessary information for the built-in [Motion Service](/services/motion/) to work with the arm, on top of the client-inaccessible and mostly static [Frame System Service](/services/frame-system/).
+
+**Find a pre-built kinematics file:**
+
+- `viam-server` will work with <file>URDF</file>  [(United Robot Description Format)]( http://wiki.ros.org/urdf) kinematics files, which are currently the standard for ROS drivers.
```
Comment (sguequierre) [inline docs/extend/modular-resources/examples/custom-arm.md:38]:
```suggestion
- `viam-server` will work with <file>URDF</file>  [(United Robot Description Format)]( https://wiki.ros.org/urdf) kinematics files, which are currently the standard for ROS drivers.
```

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 17]
```diff
@@ -0,0 +1,250 @@
...
+
+If you have a robot arm that is not already supported by the RDK, create a module that provides a customized model for your arm to [program](/program/) and control it with the [arm API](/components/arm/#api), or use it with [services](/services/) like [Motion](/services/motion/), just as you would with a built-in model.
+
+See [Modular Resources](/extend/modular-resources/) for more information.
+
+## Get your arm's kinematics file
+
+The way arms move through space is more complicated than Viam's other [components](/components/).
+Because of this, an arm, unlike other components, requires a [kinematic configuration file](/internals/kinematic-chain-config/) describing its geometry.
+This provides necessary information for the built-in [Motion Service](/services/motion/) to work with the arm, on top of the client-inaccessible and mostly static [Frame System Service](/services/frame-system/).
```
Comment (sguequierre) [inline docs/extend/modular-resources/examples/custom-arm.md:34]:
```suggestion
This provides the necessary information for the built-in [Motion Service](/services/motion/) to work with the arm.
```

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 18]
```diff
@@ -0,0 +1,250 @@
...
+
+- Follow the instructions on [Configure Complex Kinematic Chains](/internals/kinematic-chain-config/) to write a file detailing the geometry of your arm.
+  - Use the [Spatial Vector Algebra (SVA)](/internals/kinematic-chain-config/#kinematic-parameters) kinematic parameter type.
+  - Define the parameters in a </file>.json</file> file.
+  - Follow the Frame System's guide to [Configure a Reference Frame](/services/frame-system/frame-config/) when working out the relative [orientations](/internals/orientation-vector/) of the `links` on your arm.
+- View examples of the SVA and JSON format in Viam's built-in arm drivers on [GitHub](https://github.com/viamrobotics/rdk/blob/main/components/arm).
+
+Create a new directory.
+Give it the name your want you call your custom {{< glossary_tooltip term_id="model" text="model" >}} of arm.
+Save this file as <file>your-model-name.json</file> inside of this directory.
```
Comment (sguequierre) [inline docs/extend/modular-resources/examples/custom-arm.md:51]:
```suggestion
Save the JSON as <file>your-model-name.json</file> inside of this directory.
```

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 19]
```diff
@@ -0,0 +1,250 @@
...
+- `viam-server` will work with <file>URDF</file>  [(United Robot Description Format)]( https://wiki.ros.org/urdf) kinematics files, which are currently the standard for ROS drivers.
+You can find <file>URDF</file> "robot descriptions" for many industrial robot arm models on GitHub that are compatible with the Viam platform.
+
+**Create your own kinematics file:**
+
+- Follow the instructions on [Configure Complex Kinematic Chains](/internals/kinematic-chain-config/) to write a file detailing the geometry of your arm.
+  - Use the [Spatial Vector Algebra (SVA)](/internals/kinematic-chain-config/#kinematic-parameters) kinematic parameter type.
+  - Define the parameters in a </file>.json</file> file.
+  - Follow the Frame System's guide to [Configure a Reference Frame](/services/frame-system/frame-config/) when working out the relative [orientations](/internals/orientation-vector/) of the `links` on your arm.
+- View examples of the SVA and JSON format in Viam's built-in arm drivers on [GitHub](https://github.com/viamrobotics/rdk/blob/main/components/arm).
```
Comment (sguequierre) [inline docs/extend/modular-resources/examples/custom-arm.md:47]:
```suggestion
  You can view existing examples of the SVA and JSON format in Viam's [built-in arm drivers](https://github.com/viamrobotics/rdk/blob/main/components/arm).
```

File: docs/extend/modular-resources/examples/custom-arm.md [hunk 20]
```diff
@@ -0,0 +1,250 @@
...
+Create a new directory.
+Give it the name your want you call your custom {{< glossary_tooltip term_id="model" text="model" >}} of arm.
+Save the JSON as <file>your-model-name.json</file> inside of this directory.
+While completing the following step, make sure to save any new files that you generate inside of this same directory.
+
+## Create a custom arm model as a modular resource
+
+To create a custom arm model, code a module in Python with the module support libraries provided by [Viam's SDKs](/program/apis/):
+
+1. [Code a new resource model](#code-a-new-resource-model) implementing all methods the Viam RDK requires in the API definition of its built-in subtype (for example, `rdk:component:arm`).
```
Comment (sguequierre) [inline docs/extend/modular-resources/examples/custom-arm.md:58]:
```suggestion
1. [Code a new resource model](#code-a-new-resource-model) implementing all methods the Viam RDK requires in the API definition of its built-in {{< glossary_tooltip term_id="model" text="models" >}}, `rdk:component:arm`.
Access the source code for the arm component's API definition on [GitHub](https://github.com/viamrobotics/rdk/blob/main/components/arm/arm.go).
```
